packages/venia-ui/lib/components/Main/__tests__/main.spec.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import createTestInstance from '@magento/peregrine/lib/util/createTestInstance';
+import Main from '../main';
+
+jest.mock('../../Header/header', () => 'Header');
+jest.mock('../../Footer/footer', () => 'Footer');
+jest.mock('../../../classify');
+
+test('it renders correctly when masked', () => {
+    const { root } = createTestInstance(<Main isMasked={true} />);
+
+    expect(root.findByProps({ className: 'root_masked' })).toBeTruthy();
+    expect(root.findByProps({ className: 'page_masked' })).toBeTruthy();
+});
+
+test('sets different classnames when not masked', () => {
+    const { root } = createTestInstance(<Main isMasked={false} />);
+
+    expect(root.findByProps({ className: 'root' })).toBeTruthy();
+    expect(root.findByProps({ className: 'page' })).toBeTruthy();
+});
```

Comment:
I know there's some debate about the value of snapshot tests, but I feel like we currently use them to detect breaking changes. I'm not locked into that mindset so I would consider either a snapshot test here, or a simple test to verify the `children` prop to rendered.

Reply:
For what it's worth I suggested that @anthoula use these direct props assertions rather than snapshots. Your request about rendering the children is still valid though!

Reply:
Nice! Worth talking about at next peer review/team time then. Snapshot tests can serve a purpose, but we should probably not call them unit tests.

Reply:
Yep, originally had snapshot assertions, as that seems to be a current convention widely in use. Added an additional assertion for rendered children